### ISSUE TITLE
Hide the android notification from lock screen and notification bar

### ISSUE THREAD

**Issue by iam9827** [mendhak/gpslogger#491]
This request is similar to #444, however, I understand that the persistent Android notification cannot be removed entirely, as Android may then kill the task.
**Instead**, I would like the icon in the notification bar to be hidden, and for the notification to not show up on the lock screen. However,  I do understand that the notification must remain in the notification pull-down shade.
I **know** it is possible to hide the notification from the lock screen and the notification bar because a similar app that I use, [OwnTracks](https://play.google.com/store/apps/details?id=org.owntracks.android&hl=en), does exactly this. They are another open-source location tracking app, and their code is [here](https://github.com/owntracks). 

To show you how OwnTracks treats the notification compared to GPSLogger, I took a few screen shots here:
![screenshot_20161217-164549](https://cloud.githubusercontent.com/assets/24629094/21290836/773d04d0-c47e-11e6-80c5-a4ae1bc4c339.png)

![screenshot_20161217-164618](https://cloud.githubusercontent.com/assets/24629094/21290835/773cf2b0-c47e-11e6-9bcb-09a79f6b1765.png)

![screenshot_20161217-164538](https://cloud.githubusercontent.com/assets/24629094/21290834/773cb8ae-c47e-11e6-8260-15b1eb4c68a7.png)

In all screenshots, both OwnTracks and GPSLogger are running, however only the GPSLogger notification shows up on the lock screen and the notification bar. Notice how both notifications must show up in the pull-down shade, since there is no way around that (at least that I know of). 

I would love if this request was implemented in a future build of GPSLogger, as it is quite annoying having the notification show up in the lock screen and the notification bar. Thanks!

**Comment by mendhak** [comment]
Thanks for nice description of the problem.This can be done by setting [VISIBILITY_SECRET and PRIORITY_MIN](https://developer.android.com/reference/android/app/Notification.html#visibility) when creating the notification.   I'm mulling whether this should be default behavior.

**Comment by iam9827** [comment]
If not default behavior (which is understandable for people who run the logger for specific periods of time, as opposed to having it run constantly in the background for home automation like me), maybe it could be in the General Options section as a toggle, like how the 'Hide Notification Buttons' toggle is. Not sure if this is possible, but hopefully!

**Comment by mendhak** [comment]
I've removed GPSLogger from showing up in lock screen.

And like you suggested I added a toggle in General Options.  That will determine if an icon appears in the status bar.  The default is - it will appear in the status bar.  

APK with this change is ready if you want to test it out now.  If you want to wait for market release that's fine too no problem. 

[APK](https://dl.dropboxusercontent.com/u/51905631/gpslogger-82-issue491.apk) ([PGP signature](https://dl.dropboxusercontent.com/u/51905631/gpslogger-82-issue491.apk.asc)|[Checksum](https://dl.dropboxusercontent.com/u/51905631/gpslogger-82-issue491.apk.SHA256))
